stan/math/prim/fun/serializer.hpp

```diff
@@ -94,8 +96,8 @@ struct deserializer {
    */
   template <typename U, require_std_vector_t<U>* = nullptr,
             require_not_st_complex<U>* = nullptr>
-  typename stan::math::promote_scalar_type<T, U>::type read(const U& x) {
-    typename stan::math::promote_scalar_type<T, U>::type y;
+  promote_scalar_t<T, U> read(const U& x) {
+    promote_scalar_t<T, U> y;
     y.reserve(x.size());
```

Comment:
You didn't do this, but should these reserves be `y.reserve(y.size() + x.size())`?

Reply:
Not in this case, since the aim is to return a container of the same size as `x`